src/lib.rs

```diff
@@ -2505,6 +2513,92 @@ pub struct RenameParams {
     pub new_name: String,
 }
 
+#[derive(Debug, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DocumentColorParams {
+    /// The text document
+    pub text_document: TextDocumentIdentifier,
+}
+
+#[derive(Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ColorInformation {
+    /**
+     * The range in the document where this color appears.
+     */
+    pub range: Range,
+    /**
+     * The actual color value for this color range.
+     */
+    pub color: Color
+}
+
+#[derive(Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Color {
+    /**
+     * The red component of this color in the range [0-1].
+     */
+    pub red: f64,
+    /**
+     * The green component of this color in the range [0-1].
+     */
+    pub green: f64,
+    /**
+     * The blue component of this color in the range [0-1].
+     */
+    pub blue: f64,
+    /**
+     * The alpha component of this color in the range [0-1].
+     */
+    pub alpha: f64
+}
+
+#[derive(Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ColorPresentationParams {
+    /**
+     * The text document.
+     */
+    pub text_document: TextDocumentIdentifier,
+
+    /**
+     * The color information to request presentations for.
+     */
+	pub color: Color,
```

Comment:
Formatting seems off here, if you can run this through rustfmt that would be great!